website/static/js/profile.js

```diff
@@ -664,28 +750,62 @@ ListViewModel.prototype.unserialize = function(data) {
 };
 
 ListViewModel.prototype.serialize = function() {
+    var self = this;
     var contents = [];
-    if (this.contents().length !== 0 && typeof(this.contents()[0].serialize() !== undefined)) {
-        for (var i=0; i < this.contents().length; i++) {
-            contents.push(this.contents()[i].serialize());
+    
+    if (self.contents().length !== 0 && typeof(self.contents()[0].serialize() !== undefined)) {
```

Comment:
This line wasn't added by you, but I think it contains a bug.

`typeof(self.contents()[0].serialize() !== undefined))`

The parentheses are unbalanced, and `undefined` is a literal - `typeof` returns a string. I believe this was the intent:

`typeof(self.contents()[0].serialize()) !== "undefined"`